### PR TITLE
chore(trie): hold direct reference to hashed accounts in cursor

### PR DESCRIPTION
**Pull request by rkrasiuk**
## Description

Hold direct reference only to hashed accounts in the cursor.

### CODE REVIEW

**Comment by gakonst** [comment]
what was the motivation for this again?

**Comment by rkrasiuk** [comment]
@gakonst this is preparation for https://github.com/paradigmxyz/reth/pull/9079 which will make both existing in memory cursors and https://github.com/paradigmxyz/reth/pull/8199 cleaner

File: crates/trie/trie/src/hashed_cursor/post_state.rs [hunk 1]
```diff
@@ -40,8 +40,8 @@ impl<'a, CF: HashedCursorFactory> HashedCursorFactory for HashedPostStateCursorF
 pub struct HashedPostStateAccountCursor<'b, C> {
     /// The database cursor.
     cursor: C,
-    /// The reference to the in-memory [`HashedPostStateSorted`].
-    post_state: &'b HashedPostStateSorted,
+    /// The reference to the in-memory [`HashedAccountsSorted`].
+    post_state: &'b HashedAccountsSorted,
```
Comment (mattsse) [inline crates/trie/trie/src/hashed_cursor/post_state.rs:44]: unclear naming

**Review by mattsse — CHANGES_REQUESTED** [review]
unchanged `post_state` naming is slightly confusing

File: crates/trie/trie/src/hashed_cursor/post_state.rs (hunk 1 above)
Comment (rkrasiuk) [reply crates/trie/trie/src/hashed_cursor/post_state.rs:44]: agreed, fixing

Comment (rkrasiuk) [reply crates/trie/trie/src/hashed_cursor/post_state.rs]: @mattsse fixed

File: crates/trie/trie/src/hashed_cursor/post_state.rs [hunk 2]
```diff
@@ -22,7 +22,7 @@ impl<'a, CF: HashedCursorFactory> HashedCursorFactory for HashedPostStateCursorF
 
     fn hashed_account_cursor(&self) -> Result<Self::AccountCursor, reth_db::DatabaseError> {
         let cursor = self.cursor_factory.hashed_account_cursor()?;
-        Ok(HashedPostStateAccountCursor::new(cursor, self.post_state))
+        Ok(HashedPostStateAccountCursor::new(cursor, &self.post_state.accounts))
```
Comment (emhane) [inline crates/trie/trie/src/hashed_cursor/post_state.rs:25]: this combined with https://github.com/paradigmxyz/reth/pull/9077 makes it possible to walk over accounts and storage tables in parallel, right?

**Review by emhane — APPROVED** [review]

**Review by mattsse — APPROVED** [review]

File: crates/trie/trie/src/hashed_cursor/post_state.rs (hunk 2 above)
Comment (rkrasiuk) [reply crates/trie/trie/src/hashed_cursor/post_state.rs:25]: we already do that by holding an `Arc` over `HashedPostState`. this just allows some code clean up

**Pull request merged**